auditbeat/docs/modules/file_integrity.asciidoc

```diff
@@ -0,0 +1,120 @@
+////
+This file is generated! See scripts/docs_collector.py
+////
+
+[id="{beatname_lc}-module-file_integrity"]
+== File Integrity Module
+
+The `file_integrity` module sends events when a file is changed (created,
+updated, or deleted) on disk. The events contain file metadata and hashes.
+
+The module is implemented for Linux, macOS (Darwin), and Windows.
+
+[float]
+=== How it works
+
+This module uses features of the operating system to monitor file changes in
+realtime. When the module starts it creates a subscription with the OS to
+receive notifications of changes to the specified files or directories. Upon
+receiving notification of a change the module will read the file's metadata
+and the compute a hash of the file's contents.
+
+At startup this module will perform an initial scan of the configured files
+and directories to generate baseline data for the monitored paths and detect
+changes since the last time it was run. It uses locally persisted data in order
+to only send events for new or modified files.
+
+The operating system features that power this feature are as follows.
+
+* Linux - `inotify` is used, and therefore the kernel must have inotify support.
+Inotify was initially merged into the 2.6.13 Linux kernel.
+* macOS (Darwin) - Uses the `FSEvents` API, present since macOS 10.5. This API
+coalesces multiple changes to a file into a single event. {beatname_uc} translates
+this coalesced changes into a meaningful sequence of actions. However,
+in rare situations the reported events may have a different ordering than what
+actually happened.
+* Windows - `ReadDirectoryChangesW` is used.
+
+The file integrity module should not be used to monitor paths on network file
+systems.
+
+[float]
+=== Configuration options
+
+This module has some configuration options for tuning its behavior. The
+following example shows all configuration options with their default values for
+Linux.
+
+[source,yaml]
+----
+- module: file_integrity
+  paths:
+  - /bin
+  - /usr/bin
+  - /sbin
+  - /usr/sbin
+  - /etc
+  scan_at_start: true
```

Comment:
This section implies that `paths` has a default value, but if you don't declare it yourself, you get:

```
2017/12/22 07:09:57.004627 beat.go:644: CRIT Exiting: 1 error: 1 error: missing required field accessing 'auditbeat.modules.1.paths' (source:'auditbeat.yml')
Exiting: 1 error: 1 error: missing required field accessing 'auditbeat.modules.1.paths' (source:'auditbeat.yml')
```